bapsflib/_hdf/maps/controls/tests/fauxnixyz.py

```diff
@@ -0,0 +1,311 @@
+        self.configs['config01']['dset name'] = dset_name
+
+    def _add_motionlist_groups(self):
+        """Add motion list groups"""
+        # determine possible data point arrangements for motion lists
+        # 1. find divisible numbers of sn_size
+        # 2. find (Nx, Ny, Nz) combos for each dataset
+        sn_size_for_ml = []
+        NN = []
+        if self.knobs.n_motionlists == 1:
+            # set shot number size per for each motion list
+            sn_size_for_ml.append(self.knobs.sn_size)
+        else:
+            # set shot number size per for each motion list
+            sn_per_ml = int(math.floor(
+                self.knobs.sn_size / self.knobs.n_motionlists))
+            sn_remainder = (self.knobs.sn_size
+                            - ((self.knobs.n_motionlists - 1)
+                               * sn_per_ml))
+            sn_size_for_ml.extend([sn_per_ml]
+                                  * (self.knobs.n_motionlists - 1))
+            sn_size_for_ml.append(sn_remainder)
+
+        # build NN of each motion list
+        for i in range(self.knobs.n_motionlists):
+            # find divisible numbers
+            sn_div = []
+            for j in range(sn_size_for_ml[i]):
+                if sn_size_for_ml[i] % (j + 1) == 0:
+                    sn_div.append(j + 1)
+
+            # build (Nx, Ny, Nz)
+            sn_div_index = random.randint(0, len(sn_div) - 1)
+            Nx = sn_div[sn_div_index]
+            sn_div_index = random.randint(0, len(sn_div) - 1) # rerun RNG
+            Ny = sn_div[sn_div_index]
+            Nz = int(sn_size_for_ml[i] / (Nx*Ny) )
+            NN.append((Nx, Ny, Nz))
+
+        # add motionlist sub-groups
+        # - define motionlist names
+        # - create motionlist group
+        # - define motionlist group attributes
+        for i in range(self.knobs.n_motionlists):
+            # define motionlist name
+            ml_name = 'ml-{:04}'.format(i + 1)
+            self._ml.append({'name': ml_name,
+                             'size': sn_size_for_ml[i]})
+            # self._motionlist_names.append(ml_name)
+
+            # create motionlist group
+            ml_gname = ml_name
+            self.create_group(ml_gname)
+
+            # set motionlist attributes
+            self[ml_gname].attrs.update({
+                'z_port': np.float64(35.0),
+                'x0': np.float64(0.0),
+                'Nx': np.float64(NN[i][0]),
+                'dx': np.float64(0.5),
+                'y0': np.float64(0.0),
+                'Ny': np.float64(NN[i][1]),
+                'dy': np.float64(0.5),
+                'z0': np.float64(0.0),
+                'Nz': np.float64(NN[i][2]),
+                'dz': np.float64(0.5),
+                'fan_XYZ':
+                    np.bytes_('FALSE') if bool(i % 2)
+                    else np.bytes_('TRUE'),
+                'min_zdrive_steps': np.float64(16491),
+                'max_zdrive_steps': np.float64(-22833),
```

Comment:
use **NI_XYZ** real value
```suggestion
                'max_zdrive_steps': np.float64(2000000),
```